### PR TITLE
feat(python): add cwe-328 weak hash rules

### PR DESCRIPTION
**Pull request by didroe**
## Description

Adds Python CWE-328 rules for weak hash algorithms

## Checklist
If this is your first time contributing please [sign the CLA](https://docs.bearer.com/contributing/)

- [ ] My rule has adequate metadata to explain its use.

### CODE REVIEW

File: rules/python/lang/weak_hash_adler32.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          - variable: NAME
+            values: [adler32]
+      - either:
+          - variable: OPTIONAL_DATA_TYPE
+            detection: python_shared_lang_datatype
+            scope: result
+          - not:
+              variable: OPTIONAL_DATA_TYPE
+              detection: python_shared_lang_datatype
+              scope: result
```
Comment (elsapet) [inline rules/python/lang/weak_hash_adler32.yml:23]: IIRC the optional datatype stuff was mostly for dynamic severity, so we could remove it now that severity is static and just catch on any usage of the weak algorithms. 

Having said that, probably something we should change across all languages for consistency. What do you think?

File: rules/python/lang/weak_hash_md5.yml [hunk 2]
```diff
@@ -1,58 +0,0 @@
...
-      ```python
-      hashlib.md5('password').digest() # unsafe
-      ```
-    - **Do** opt for stronger hashing algorithms like SHA-256 to enhance security.
-      ```python
-      hashlib.sha256('password').digest()
-      ```
-  cwe_id:
-    - 328
-  id: python_lang_weak_hash_md5
```
Comment (elsapet) [inline rules/python/lang/weak_hash_md5.yml:55]: What's the impact of deleting rules?

File: rules/python/lang/weak_hash_dss.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      - variable: FUNCTION
+        detection: python_shared_lang_import1
+        scope: cursor
+        filters:
+          - variable: MODULE1
+            values: [hashlib]
+          - variable: NAME
+            values: [new]
+      - variable: ALGORITHM
+        string_regex: (?i)\Adss
```
Comment (elsapet) [inline rules/python/lang/weak_hash_dss.yml:16]: For my understanding, why don't we match on the end of the string here? 

```suggestion
        string_regex: (?i)\Adss\z
```

**Review by elsapet — APPROVED** [review]

File: rules/python/lang/weak_hash_dss.yml (hunk 3 above)
Comment (didroe) [reply rules/python/lang/weak_hash_dss.yml:16]: I think you can use `DSS1` in some cases

Comment (elsapet) [reply rules/python/lang/weak_hash_dss.yml:16]: Ah cool -- thanks 🙏

File: rules/python/lang/weak_hash_md5.yml (hunk 2 above)
Comment (didroe) [reply rules/python/lang/weak_hash_md5.yml:55]: I'm not really sure what expectations are pre-GA. I'll change it back

**Pull request merged**